hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsRestOperation.java

```diff
@@ -229,14 +232,29 @@ private void completeExecute(TracingContext tracingContext)
       }
     }
 
-    if (result.getStatusCode() >= HttpURLConnection.HTTP_BAD_REQUEST) {
+    int status = result.getStatusCode();
+    /*
+      If even after exhausting all retries, the http status code has an
+      invalid value it qualifies for InvalidAbfsRestOperationException.
+      All http status code less than 1xx range are considered as invalid
+      status codes.
+     */
+    if (status < HTTP_CONTINUE) {
+      throw new InvalidAbfsRestOperationException(null, retryCount);
```

Comment:
this is clearly something odd. the status code should be included in the exception to help debug

Reply:
Hardcoded while calling the constructor of the parent since the status code for invalid abfsrestoperation is always -1.